File: BankManagementSystem-backend/src/main/java/com/bank/model/Customer.java
package com.bank.model;

import java.util.Date;
import java.util.List;

import jakarta.persistence.CascadeType;
import jakarta.persistence.Entity;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;
import jakarta.persistence.OneToMany;
import lombok.Data;

@Data
@Entity
public class Customer 
{
	@Id
	@GeneratedValue(strategy = GenerationType.IDENTITY)
	private Long customerId;
	private String name;
	private Date dateofbirth;
	private String email;
	private Long contact;
	private String address;
	private String city;
	private String state;
	private Integer pincode;
	
	
	 @OneToMany(mappedBy = "customer", cascade = CascadeType.ALL)
	 private List<Account> accounts;
	
	
}

File: BankManagementSystem-frontend/BankManagementSystem-Angular/src/app/customer.ts
export class Customer {
    customerId: number;
    name: string;
    dateofbirth: Date;
    email:String;
    contact: string;
    address: string;
    city: string;
    state: string;
    pincode: string;
  
    constructor(
      customerId: number,
      name: string,
      dateofbirth: Date,
      email:String,
      contact: string,
      address: string,
      city: string,
      state: string,
      pincode: string
    ) {
      this.customerId = customerId;
      this.name = name;
      this.dateofbirth = dateofbirth;
      this.email=email;
      this.contact = contact;
      this.address = address;
      this.city = city;
      this.state = state;
      this.pincode = pincode;
    }
  }
  